sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -109,10 +109,7 @@ class Analyzer(
     this(catalog, conf, conf.optimizerMaxIterations)
   }
 
-  override protected def defaultCatalogName: Option[String] = conf.defaultV2Catalog
-
-  override protected def lookupCatalog(name: String): CatalogPlugin =
-    throw new CatalogNotFoundException("No catalog lookup function")
+  override val catalogManager: CatalogManager = new CatalogManager(conf)
```

Comment:
I thought that this was going to be part of SessionState and not Analyzer. Why not SessionState?

Reply:
Because it's only needed by the analyzer.

Alternatively, we can put `CatalogManager` in `BaseSessionStateBuilder`, and pass it to analyzer when we create analyzer in `BaseSessionStateBuilder`.

Technically, it's already in `SessionState` because `SessionState` holds the analyzer.

Reply:
I can add a `SessionState.catalogManager` which calls `analyzer.catalogManager`, to make it easier to access the catalog manager. What do you think?

Reply:
Let's put CatalogManager in SessionState and pass it into the analyzer. That's a better place for the catalog manager.

Reply:
> Let's put CatalogManager in SessionState and pass it into the analyzer

This is not doable. The analyzer is created in `BaseSessionStateBuilder`. So there are 2 options:
1. create the `CatalogManager` in `BaseSessionStateBuilder`, and pass it to the analyzer.
2. create the `CatalogManager` in the analyzer, and put an accessor of it in `SessionState`.

I think 2 is better.

Reply:
Okay, let's do #2 then.